src/device-manager/java/WeaveDeviceManager-JNI.cpp

```diff
@@ -3294,28 +3421,11 @@ WEAVE_ERROR N2J_NetworkInfo(JNIEnv *env, const NetworkInfo& inNetworkInfo, jobje
 
 WEAVE_ERROR N2J_NetworkInfoArray(JNIEnv *env, const NetworkInfo *inArray, uint32_t inArrayLen, jobjectArray& outArray)
 {
-    WEAVE_ERROR err = WEAVE_NO_ERROR;
-    jobject elem = NULL;
-
-    outArray = env->NewObjectArray(inArrayLen, sNetworkInfoCls, NULL);
-    VerifyOrExit(outArray != NULL, err = WEAVE_ERROR_NO_MEMORY);
-
-    for (uint32_t i = 0; i < inArrayLen; i++)
-    {
-        err = N2J_NetworkInfo(env, inArray[i], elem);
-        SuccessOrExit(err);
-
-        env->ExceptionClear();
-        env->SetObjectArrayElement(outArray, i, elem);
-        VerifyOrExit(!env->ExceptionCheck(), err = WDM_JNI_ERROR_EXCEPTION_THROWN);
-
-        env->DeleteLocalRef(elem);
-        elem = NULL;
-    }
-
-exit:
-    env->DeleteLocalRef(elem);
-    return err;
+    return N2J_GenericArray(env, inArrayLen, sNetworkInfoCls, outArray,
```

Comment:
Neat!  A clarification: Is the primary reason for `N2J_GenericArray` for being a template so that it may take both closures and functions?  Could it not work just with closures without being a template?

Reply:
Discussed off-line